src/transaction_processor.py

```diff
@@ -84,71 +97,175 @@ def process(self, start_block: int) -> None:
 
     def process_single_transaction(
         self, tx_hash: str, auction_id: int, block_number: int
-    ):
-        """Function processes a single tx to find imbalances, prices."""
+    ) -> None:
+        """Function processes a single tx to find imbalances, fees, prices including writing to database."""
+        self.log_message = []
         try:
-            token_imbalances = self.imbalances.compute_imbalances(tx_hash)
-        except Exception as e:
-            logger.error(f"Failed to compute imbalances for transaction {tx_hash}: {e}")
-            return
+            # Compute Raw Token Imbalances
+            if self.process_imbalances:
+                token_imbalances = self.process_token_imbalances(
+                    tx_hash, auction_id, block_number
+                )
 
-        log_message: list[str] = []
-        log_message.append(f"Token Imbalances on {self.chain_name} for tx {tx_hash}:")
-        for token_address, imbalance in token_imbalances.items():
-            # write imbalance to table if it's non-zero
-            if imbalance != 0:
-                self.db.write_token_imbalances(
-                    tx_hash, auction_id, block_number, token_address, imbalance
+            # Compute Fees
+            if self.process_fees:
+                protocol_fees, network_fees = self.process_fees_for_transaction(
+                    tx_hash, auction_id, block_number
                 )
-                log_message.append(f"Token: {token_address}, Imbalance: {imbalance}")
 
-        protocol_fees, network_fees = batch_fee_imbalances(HexBytes(tx_hash))
-        self.handle_fees(protocol_fees, network_fees, auction_id, block_number, tx_hash)
-        slippage = calculate_slippage(token_imbalances, protocol_fees, network_fees)
+            # Compute Prices
+            if self.process_prices:
+                prices = self.process_prices_for_tokens(
+                    token_imbalances, protocol_fees, network_fees, block_number, tx_hash
+                )
+
+            # Write to database iff no errors in either computations
+            if self.process_imbalances or self.process_fees or self.process_prices:
```

Comment:
In order to avoid indenting too much, we could instead write

```
if (not self.process_imbalances) and (not self.process_fees) and (not self.process_prices):
    return
```